pkg/cri/relay/relay.go

```diff
@@ -134,6 +150,34 @@ func (r *relay) Server() server.Server {
 	return r.server
 }
 
+func (r *relay) dialNotify(socket string, perm os.FileMode, uid int, gid int, err error) {
+	if err != nil {
+		r.Error("failed to determine permissions/ownership of client socket %q: %v",
+			socket, err)
+		return
+	}
+
+	// we ignore image socket properties, unless it is the only socket we have
+	if r.options.ImageSocket != r.options.RuntimeSocket {
+		if socket != r.options.RuntimeSocket && r.options.RuntimeSocket != client.DontConnect {
+			return
+		}
+	}
+
+	r.perm = perm
+	r.uid = uid
+	r.gid = gid
+
+	if r.started {
```

Comment:
So if the relay isn't started, we don't `chown`? Shouldn't we try to `chown` precisely when the relay is not running to avoid any race?

Reply:
If you use the stock golang net.Listener/net.Dial() family of functions to create a unix-domain listening socket and then you change its ownership/permissions, then you simply cannot avoid the race condition. While the unix system call interface properly separates socket creation, binding and listening, the golang net abstraction hides creating the socket and listening on it behind a single function call.